packages/askui-nodejs/src/utils/http/http-client-got.ts

```diff
@@ -11,14 +17,83 @@ export class HttpClientGot {
 
   private askuiGot: Got;
 
+  urlsToRetry: string[] = [];
+
   constructor(
     readonly token?: string,
     readonly customHeaders?: Record<string, string>,
     private readonly cookies: Record<string, string> = {},
     readonly proxyAgents?: { http: http.Agent; https: https.Agent },
   ) {
     this.initHeaders(token, customHeaders);
-    this.askuiGot = got.extend(proxyAgents ? { agent: proxyAgents } : {});
+    const gotExtendOptions = this.buildGotExtendOptions(proxyAgents);
+    this.askuiGot = got.extend(gotExtendOptions);
+  }
+
+  private buildGotExtendOptions(
+    proxyAgents?: { http: http.Agent; https: https.Agent } | undefined,
+  ): ExtendOptions {
+    const gotExtendOptions: ExtendOptions = {
+      retry: {
+        limit: 5,
+        methods: ['POST', 'GET', 'PUT', 'HEAD', 'DELETE', 'OPTIONS', 'TRACE'],
+        statusCodes: [408, 413, 429, 500, 502, 503, 504, 521, 522, 524],
+        errorCodes: [
+          'ETIMEDOUT',
+          'ECONNRESET',
+          'EADDRINUSE',
+          'ECONNREFUSED',
+          'EPIPE',
+          'ENOTFOUND',
+          'ENETUNREACH',
+          'EAI_AGAIN',
+        ],
+        calculateDelay: ({
+          attemptCount,
+          retryOptions,
+          error,
+          computedValue,
+        }) => {
+          if (
+            attemptCount > retryOptions.limit
+            || !this.shouldRetryOnError(error)
+          ) {
+            return 0;
+          }
+
+          if (
+            error.response !== undefined
+            && error.response.headers['retry-after'] === undefined
+          ) {
+            return Math.min(
+              1000 * 2 ** (attemptCount - 1) + Math.random() * 100,
+              Number.MAX_SAFE_INTEGER,
+            );
+          }
```

Comment:
should we log a warning after each retry?  
and can this cause a timeout error?

Reply:
Thx for pointing out the missing logging. I add it but on a debug level as I would like to hide that as long as the last request succeeds. 

No timeout error in the integration tests but could happen potentially which I think is fine.